<launch>
  
  <!-- Arguments -->
  <arg name="open_rviz" default="true"/>
  <arg name="map_file" default="$(find testbot)/maps/map.yaml"/>

  <group ns="big_chicken">
    <arg name="init_x_" default="0.3" />
    <arg name="init_y_" default="1.5" /> <!--2.68-->
    <arg name="init_yaw_" default="0.0" />

    <node pkg="simple_version_navigation" type="pose_tracker_2" name="pose_tracker" output="screen" />
    <param name="odom_name" value="odom_big_chicken"/>
    <param name="base_footprint_name" value="base_footprint_big_chicken"/>
    <rosparam file="$(find simple_version_navigation)/param/pose_tracker_params.yaml" command="load" />
    <!-- <rosparam file="$(find simple_version_navigation)/param/pose_tracker_params_big.yaml" command="load" /> -->

    
    <!-- x y z y p r--> 
    <node pkg="tf" type="static_transform_publisher" name="map_odom_broadcaster"  args="0 0 0 0 0 0 map odom_big_chicken 0.1" />


    <node pkg="testbot" name = "odometry_test" type = "odometry_test"  output="screen">
      <param name="init_x" value="$(arg init_x_)"/>
      <param name="init_y" value="$(arg init_y_)"/>
      <param name="init_yaw" value="$(arg init_yaw_)"/>
      <param name="odom_name" value="odom_big_chicken"/>
      <param name="base_footprint_name" value="base_footprint_big_chicken"/>
    </node>

    <node pkg="testbot" name = "planner_to_ST" type = "planner_to_ST"  output="screen">
      <param name="init_x" value="$(arg init_x_)"/>
      <param name="init_y" value="$(arg init_y_)"/>
      <param name="init_yaw" value="$(arg init_yaw_)"/>
    </node>

    <!-- goal -->
    <node pkg="testbot" name = "goal_pose_multipoint" type = "goal_pose_multipoint"  output="screen">
      <rosparam param="goal_list">[[0.3,1.5,0.0],[1.5,1.5,0.0]]</rosparam>
      <param name="goal_loop" value="true" />
    </node>
  </group>
  <!-- rviz -->
  <!-- <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find simple_version_navigation)/rviz/pose_tracker.rviz"/>
  </group> -->

  <!-- Map server -->
  <!-- <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/> -->

</launch>